### PR TITLE
PXB-2258 document --lock-ddl and --lock-ddl-per-table (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
new file:   source/advanced/locks.rst
modified:   source/faq.rst
modified:   source/index.rst

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+1. Parse and copy all redo logs after the checkpoint mark
+
+2. Fork a dedicated thread to continue following new redo log entries
+
+3. List the tablespaces required to copy
+
+4. Iterate through the list. The following steps occur with each listed tablespace:
+
+    a. Query INFORMATION_SCHEMA.INNODB_SYS_TABLES to find which tables belong
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst]:
```suggestion
    a. Query INFORMATION_SCHEMA.INNODB_TABLES to find which tables belong
```

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+event is found by the redo follow thread, the backup continues and assumes
+the backup is safe because the MDL protects tablespaces already copied and
+the MLOG_INDEX_LOAD event is for a tablespace that is not copied.
+
+These assumptions may not be correct and may lead to inconsistent backups.
+
+``--lock-ddl-per-table`` redesign
+----------------------------------
+
+Implemented in |Percona XtraBackup| version 2.4.21, the
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst]:
```suggestion
Implemented in |Percona XtraBackup| version 8.0.22, the
```

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+The following improvements have been added:
+
+* If the ``.ibd`` file belongs to a temporary table, the ``SELECT`` query
+is skipped.
+
+* For a FullText Index, an MDL is acquired on the base table.
+
+* A ``SELECT`` query that acquires an MDL does not retrieve any data.
+
+
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst:77]: Add a note that `lock-ddl-per-table` is deprecated on 8.0 and users should use `--lock-ddl instead`

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst]: Fixed.

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst]: Fixed.

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+The backup process may encounter a redo log event, generated by the bulk load
+operations, which notifies backup tools that data file changes have been
+omitted from the redo log. This event is an ``MLOG_INDEX_LOAD``. If this
+event is found by the redo follow thread, the backup continues and assumes
+the backup is safe because the MDL protects tablespaces already copied and
+the MLOG_INDEX_LOAD event is for a tablespace that is not copied.
+
+These assumptions may not be correct and may lead to inconsistent backups.
+
+``--lock-ddl-per-table`` redesign
```
Comment (patrickbirch) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst:40]: Should I remove this section since --lock-ddl-per-table is deprecated?

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst:40]: No, the same redesign applies to 8.0, but we are encouraging users to use --lock-ddl instead. On 2.4 lock-ddl was only available on PS and on 8.0 it is available in upstream and PS.

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]
Depecrated warning

**Comment by patrickbirch** [comment]
changed to warning

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**